exemplars_parent/exemplars/src/main/java/io/prometheus/client/exemplars/api/Exemplar.java

```diff
@@ -0,0 +1,106 @@
+package io.prometheus.client.exemplars.api;
+
+import java.util.Arrays;
+
+/**
+ * Immutable data class holding an Exemplar.
+ */
+public class Exemplar {
+
+  public static final String TRACE_ID = "trace_id";
+  public static final String SPAN_ID = "span_id";
+
+  private final String[] labels;
+  private final double value;
+  private final long timestampMs;
+
+  /**
+   * Create an Exemplar without a timestamp
+   *
+   * @param value  the observed value
+   * @param labels name/value pairs. Expecting an even number of strings. The combined length of the label names and
+   *               values must not exceed 128 UTF-8 characters. Neither a label name nor a label value may be null.
+   */
+  public Exemplar(double value, String... labels) {
```

Comment:
Passing in labels should be consistent with how it's done in Info, so a map should also be supported.

Reply:
I moved the `Exemplar` and all classes that depend on it to `simpleclient`. Now only the bridges to the tracing libraries live in separate modules. I agree that it should be safe to have simpleclient instrumentation in place, and that dependencies should be avoided. I hope this is the case with the current implementation: `simpleclient` does not have any runtime dependencies. OpenTelemetry is available we will load the corresponding `SpanContextSupplier`, if not we won't.

We could disable exemplars by default and have the user explicitly enable them with a `ExemplarConfig.enableExemplars()` call. Apart from that I don't see how to improve this. We could load `SpanContextSuppliers` dynamically through a class loader, but that would yield to the same optional dependency.

I added an `Exemplar` constructor with a `Map` parameter to be consistent with `Info`.